x-pack/plugins/security_solution/public/management/components/console/components/console_manager/components/console_exit_modal_link.tsx

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useMemo, useCallback, memo } from 'react';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { EuiText, EuiLink } from '@elastic/eui';
+import { useAppUrl } from '../../../../../../common/lib/kibana/hooks';
+import { getEndpointDetailsPath } from '../../../../../common/routing';
+import { useEndpointSelector } from '../../../../../pages/endpoint_hosts/view/hooks';
+import { CONSOLE_EXIT_MODAL_INFO } from '../translations';
+import { useNavigateByRouterEventHandler } from '../../../../../../common/hooks/endpoint/use_navigate_by_router_event_handler';
+import { uiQueryParams } from '../../../../../pages/endpoint_hosts/store/selectors';
+
+export const HostNameText = ({ hostName }: { hostName: string }) => (
+  <EuiText size="s" style={{ maxWidth: 120, display: 'inline-flex' }}>
+    <span className="eui-textTruncate">
+      <strong>{hostName}</strong>
+    </span>
+  </EuiText>
+);
+
+export const ConsoleExitModalActionLogLink = memo(
+  ({
+    agentId,
+    hostName,
+    onClose,
+    'data-test-subj': dataTestSubj,
+  }: {
+    agentId: string;
+    'data-test-subj'?: string;
+    hostName: string;
+    onClose: () => void;
+  }) => {
+    const { getAppUrl } = useAppUrl();
+    const { show: _, ...currentUrlQueryParams } = useEndpointSelector(uiQueryParams);
```

Comment:
There are two issues here:

1. you should not be using redux store selectors. Responder is opened from a few places, including places outside of the Management section.
2. This console should not "know" anything about "endpoint" because it is likely that it will be expanded to other actions (ex. osQuery, fleet), thus why just about everything that is "endpoint specific" is provided as an input prop.

### My suggestion:
Add a new optional Component prop to the ConsoleManager console [registration interface](https://github.com/elastic/kibana/blob/6efef0496077f4e61d49e4e43f75941ff3d98d9e/x-pack/plugins/security_solution/public/management/components/console/components/console_manager/types.ts#L13-L38) (ex. `ExitConfirmationComponent`) so that we can define this for Endpoint when we register a new console ([we open/register new consoles for Endpoint here](https://github.com/elastic/kibana/blob/f3eedfb5656196d8b48f38edcd09bd91f5baea02/x-pack/plugins/security_solution/public/management/hooks/endpoint/use_with_show_endpoint_responder.tsx#L45-L58) ). You can follow the same pattern that was already done for the other `Page**Component` props on this interface.

If this component is not defined for the registered console, then we should not show anything and exit with out a modal being shown. (or maybe we just show a generic message).

As for the URL params - case the user is on Endpoint list: I suggest using the hook `useUrlParams()` that we have for it. I don't think we have a hook specific for the Endpoint list because that page has not been refactored to remove redux.

Reply:
done 58fabe66865f7d1e624cac64bd153792ace085e8